### DISCUSSION TITLE
How can I use async function in slint-ui

### DISCUSSION THREAD

**Discussion by Decodetalkers** [slint-ui/slint#1364] [Q&A]
I haven't found async key word in the example. How can I use it ?

**Comment by tronical** [comment]
You’re right, none of our examples use async (yet). However you can look into the cargo-ui source code to find an approach of combining Slint with async: https://github.com/slint-ui/cargo-ui

**Comment by ogoffart** [comment]
Right, we should have a convinient way to run async code.
But we actually do use future in our viewer and LSP implementation

https://github.com/slint-ui/slint/blob/474eb3482ecaae146d11edfba538c96f5e06ea9a/tools/viewer/main.rs#L342-L369

Unfortunately, you can't use it as is as it uses private API, but you can replace the `i_slint_backend_selector::backend().post_event` with [`slint::invoke_from_event_loop`](https://docs.rs/slint/latest/slint/fn.invoke_from_event_loop.html#)  and that should get you a way to run futures in the slint event loop.
